web3swift/src/Contract/Statically Typed/ABIConstructorEncoder.swift

```diff
@@ -0,0 +1,55 @@
+//  Created by Rinat Enikeev on 23.04.2022.
+
+import Foundation
+import BigInt
+
+public class ABIConstructorEncoder {
```

Comment:
Construction could be seen also as a special function call, so it makes sense from the types point of view.

I think we should leverage `ABIFunctionEncoder` which is already tested instead of duplicating code here.

Can create an `enum` in `ABIFunctionEncoder` with two cases:
```swift
enum EncodeVariant {
     case call(name: String)
     case construct(bytecode: Data)
}
```

Then it's just a matter of changing `encoded` function to support both calls.

To keep API backwards compatible, we can still keep the same initializer with `name` which would just set the encoding type to `call`